data-access/EditionRepository.cs

```diff
@@ -143,15 +143,17 @@ public async Task ChangeEditionNameAsync(uint editionId, string name, UserInfo u
                     
                     // Now TrackMutation will insert the data, make all relevant changes to the owner tables and take
                     // care of main_action and single_action.
-                    var response = await _databaseWriter.WriteToDatabaseAsync(editionId, user, new List<MutationRequest>() { nameChangeRequest });
+                    var response = await _databaseWriter.WriteToDatabaseAsync(user, new List<MutationRequest>() { nameChangeRequest });
                     // Bronson: Where isn't there a response?
+                    // Itay: You get no response when you try to change to the name that is already being used.
+                    // This can happen when, after the database write begins, someone else changed it to the same name first.
```

Comment:
I think these are two cases:

1. The user hasn't specified a new name. This can be either swallowed by the server or return a 400 Bad Request.
2. The user has specified a name that's used by another edition. In this case return a 409 Conflict. We may add a check for this in the frontend, as we have all the user's editions anyway. The server sholudn't rely on frontend checks, though, and should always perform them itself.

Reply:
1. Agreed, If the user hasn't specified a name, we should return immediately with a 404. 

2. This is not the case I had in mind, there is no restriction on naming and multiple edition are allowed to have the same name, so there is never a 409 Conflict possible in this case.  The difference is between whether a change was made to the database or no change was made to the database (hence response.Count == 0).  If you try to change the name to what the edition is already named, no data will be written to the database.  In such a case, should we inform the user that the request did nothing (though it was otherwise valid, and the database does reflect the desired outcome)?  Or should we just tell them it was successful (since the state of the database is exactly what they requested)?  Other ideas?

Reply:
Do you mean I can have two different editions with exactly the same name? Isn't that going to be a problem for users? How am I going to tell the difference between the two? Obviously two users can have two editions with the same name, but one user with duplicate names? I think this is mostly a source for trouble. "I changed this and that, but I can't see these changes!" emails. Frantic emails.

As for returning an error when nothing changed as opposed to returning 200 - let's look at it from the client's perspective. I think it's better not to report an error at all - just like an UPDATE statement that doesn't change anything doesn't fail, or "x = x" doesn't throw an exception. However, this is a tiny issue, and if we return an error - that's fine too.

Reply:
Ok, I can see the value of 200 here, so maybe we just do that.  You bring up a good point when a user has more than one edition with the same name, that can be confusing (though the database technically allows it, since a "power user" could look at the editionId to differentiate).  I too think that we should prevent the average user from having two editions with the same name, you are right that they will probably confuse things.  So my question is, do we use the ScrollEditor website logic to prevent such an action entirely (you should already have the info there to know this case), do we use ScrollEditor website logic to provide a strong warning (with a note to look at the editionId), but ultimately allow the user to do it, or do we prevent this operation in the API (you would also probably block it in the website logic too)? I will email this question to Ingo too for his opinion.

Reply:
I removed the superfluous check.  We should work in conjunction with Ingo to fully resolve this issue in a later PR.